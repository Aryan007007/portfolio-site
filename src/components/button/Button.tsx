import styled from "styled-components";

// Default Button Styles
const Button = styled.button`
  height: 3rem;
  min-width: 10rem;
  width: ${props => (props.theme.screens.sm ? "100%" : "18rem")};
  border: none;
  background: ${props => props.theme.colors.secondary};
  color: white;
  border-radius: 0.25rem;
  font-size: ${props => props.theme.fontSize.button};
  cursor: pointer;
  :disabled {
    background: lightgray;
  }
`;

export { Button };
