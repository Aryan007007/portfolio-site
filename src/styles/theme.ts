export default {
  screens: {
    // max-widths in pixels
    xs: 450,
    sm: 640,
    md: 768,
    lg: 1024,
    xl: 1280,
    xxl: 1350,
    xxxl: 1780,
  },
  maxWidth: "1920px",
  fontSize: {
    fadedHeading: "400px",
    sectionHeading: {
      xxxl: "196px",
      xxl: "148px",
      lg: "96px",
      sm: "48px",
    },
    heading: { xxxl: "96px", xxl: "64px", md: "36px" },
    subHeading: { xxxl: "36px", xxl: "24px", md: "18px" },
    body1: { xxxl: "24px", xxl: "18px", md: "14px" },
    body2: { xxxl: "18px", xxl: "16px", md: "12px" },
    body3: { xxxl: "16px", xxl: "14px", md: "11px" },
  },
  colors: {
    primary: "#f2f2f2",
    secondary: "#000",
    textPrimary: "#000",
    textSecondary: "#fff",
    textHint: "#00000066",
  },
  graph: {
    hierarchyGraph: {
      fontSize: { md: 14, sm: 10 },
      rootRadius: { md: 32, sm: 20 },
      rectNode: {
        width: { md: 64, sm: 36 },
        height: { md: 32, sm: 20 },
        radius: { md: 4, sm: 2 },
      },
    },
  },
  opacity: {
    0: 0,
    25: 0.25,
    30: 0.3,
    50: 0.5,
    75: 0.75,
    100: 1,
  },
  zIndex: {
    content: "10",
  },
};
