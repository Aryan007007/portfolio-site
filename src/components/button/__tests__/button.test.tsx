import { render } from "@testing-library/react";
import "jest-styled-components";
import React from "react";
import { ThemeProvider } from "styled-components";
import theme from "../../../styles/theme";
import { Button } from "../Button";

it("matches snapshot", () => {
  jest.isolateModules(() => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Button>Hello World!</Button>
      </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
