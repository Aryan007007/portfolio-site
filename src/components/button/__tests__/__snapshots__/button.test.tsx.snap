// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  height: 3rem;
  min-width: 10rem;
  width: 100%;
  border: none;
  color: white;
  border-radius: 0.25rem;
  font-size: 1.125rem;
  cursor: pointer;
}

.c0:disabled {
  background: lightgray;
}

<button
    class="c0"
  >
    Hello World!
  </button>
</DocumentFragment>
`;
